libcasr/src/csharp.rs

```diff
@@ -133,34 +133,234 @@ impl ParseStacktrace for CSharpStacktrace {
+        }
+
+        match get_regex("at A`1[T].h[Z] (System.Func`1[TResult] a) [0x00001] in <f6b2b0ea894844dc83a96f9504d8f570>:0") {
+            Ok(re) => assert_eq!(re, re_mono),
+            Err(err) => panic!("{err}")
+        }
+
+        match get_regex("at A`1.<set_Q>g__g|1_0(Int32[] arr(((((((((( in /home/user/dotnet/2/A.cs:line 19") {
+            Ok(_) => assert!(false),
+            Err(err) => assert_eq!(err.to_string(), "Casr: Couldn't parse stacktrace line: at A`1.<set_Q>g__g|1_0(Int32[] arr(((((((((( in /home/user/dotnet/2/A.cs:line 19")
+        }
+
+        match get_regex("at Program+<>c.<Main>b__0_2 (System.Int32 i) <0x7f30488306b0 + 0x00035> in <f6b2b0ea894844dc83a96f9504d8f570#610bc057486c618efb3936233b088988>:0") {
+            Ok(re) => assert_eq!(re, re_mono),
+            Err(err) => panic!("{err}")
+        }
+
+        match get_regex("at (wrapper managed-to-native) System.Drawing.GDIPlus:GdiplusStartup (ulong&,System.Drawing.GdiplusStartupInput&,System.Drawing.GdiplusStartupOutput&)") {
+            Ok(re) => assert_eq!(re, re_mono),
+            Err(err) => panic!("{err}")
+        }
+
+        match get_regex("at Program.<Main>g__f|0_0(Func`2 d, Int32& l, Int32 m) in /home/user/dotnet/2/Program.cs:line 9") {
+            Ok(re) => assert_eq!(re, re_dotnet),
+            Err(err) => panic!("{err}")
+        }
+
+        match get_regex("at Program.Main()") {
+            Ok(re) => assert_eq!(re, re_dotnet),
+            Err(err) => panic!("{err}")
+        }
+    }
+
+    #[test]
+    fn test_csharp_parse_exception() {
+        let streams = [
+            "
+Unhandled Exception:
+System.ArithmeticException: 123
+
+321
+
+7
+ ---> System.ArgumentException: 1111 ---> System.IO.IOException: cccc
+   --- End of inner exception stack trace ---",
+            "
+Unhandled Exception:
+System.ArgumentException: 1111
+ ---> System.IO.IOException: cccc
+   --- End of inner exception stack trace ---",
+            "
+Unhandled exception. System.ArgumentException: 1111
+ ---> System.IO.IOException: cccc
+   --- End of inner exception stack trace ---",
+            "
+Unhandled exception. System.ArgumentException: 1111 ---> System.IO.IOException: cccc
+   --- End of inner exception stack trace ---"
+        ];
+
+        let exceptions = streams.map(|s| {
+            let Some(e) = CSharpException::parse_exception(s) else {
+                panic!("Couldn't get C# exception from stream: {s}");
+            };
+            e
+        });
+
+        assert_eq!(exceptions[0].short_description, "System.IO.IOException");
+        assert_eq!(exceptions[0].description, "cccc");
+        assert_eq!(exceptions[1].short_description, "System.IO.IOException");
+        assert_eq!(exceptions[1].description, "cccc");
+        assert_eq!(exceptions[2].short_description, "System.IO.IOException");
+        assert_eq!(exceptions[2].description, "cccc");
+        assert_eq!(exceptions[3].short_description, "System.ArgumentException");
```

Comment:
This test case should be parsed correctly with changes applied inside CSharpException::parse_exception

Reply:
Why it will be incorrect? As I see, it **should be** `"System.IO.IOException"` as in other cases